core/src/services/s3/backend.rs

```diff
@@ -1052,6 +1052,7 @@ impl Accessor for S3Backend {
 
         match status {
             StatusCode::NO_CONTENT => Ok(RpDelete::default()),
+            StatusCode::NOT_FOUND => Ok(RpDelete::default()), // allow 404 when deleting a non-existing object
```

Comment:
Please make it more clear that it's not a standard behavior, only some s3 alike service like gcs xml api do this.

And please add comments upper this line instead of after.

Reply:
done.